### PR TITLE
BOAC-1495: Adds Google Analytics tracking for curated groups

### PR DESCRIPTION
**Pull request by lyttam**
https://jira.ets.berkeley.edu/jira/browse/BOAC-1495

### CODE REVIEW

File: src/api/curated.ts [hunk 1]
```diff
@@ -85,10 +98,12 @@ export function removeFromCuratedGroup(groupId, sid) {
...
-      store.dispatch('curated/updateCuratedGroup', group).then(() => {
-        // TODO: implement GA tracking (BOAC-1506)
-        // googleAnalyticsService.track('Curated Cohort', 'remove_student', cohort.name, cohort.id);
-        return group;
+      store.dispatch('curated/updateCuratedGroup', group);
+      return group;
+    })
+    .then(group => {
+      event('Curated Cohort', 'remove_student', group.name, group.id, {
+        userId: store.getters['user/currentUser'].uid
```
Comment (johncrossman) [inline src/api/curated.ts]: @lyttam you'll need to change the above to `store.getters['user/user']`. In order to leverage `...mapGetters` (eg, in `UserMetadata`) I did away with 'currentUser'. Sorry!  See all five cases above.

File: src/main.ts [hunk 2]
```diff
@@ -25,6 +27,12 @@ Vue.config.productionTip = false;
 Vue.use(BootstrapVue);
 Vue.use(require('vue-lodash'));
 Vue.use(VueHighcharts);
+Vue.use(VueAnalytics, {
+  id: store.dispatch('context/loadConfig').then(response => {
+    return _.get(response, 'googleAnalyticsId');
+  }),
+  checkDuplicatedScript: true
+});
```
Comment (johncrossman) [inline src/main.ts]: Nice!

**Comment by johncrossman** [comment]
Well done, @lyttam  I'll merge once `currentUser` is amended. Thanks!

File: src/api/curated.ts (hunk 1 above)
Comment (lyttam) [reply src/api/curated.ts]: Thanks @johncrossman, amended.

**Pull request merged**